src/main/app/case/to-api-format.ts

```diff
@@ -61,7 +61,11 @@ const fields: ToApiConverters = {
   }),
   applicant2AddressPrivate: data => ({
     applicant2ContactDetailsConfidential:
-      data.applicant2AddressPrivate === YesOrNo.YES ? ConfidentialAddress.KEEP : ConfidentialAddress.SHARE,
+      data.applicant2AddressPrivate === YesOrNo.YES
+        ? ConfidentialAddress.KEEP
+        : data.applicant2AddressPrivate === null
```

Comment:
i dont see why we are checking for null?

Reply:
As discussed, the app 2 questions are all pulled through as null once app 2 is linked but because there is a handler for this particular field the field is getting set to No.